packages/coding-standard/src/Rules/PreferredRawDataInTestDataProviderRule.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Symplify\CodingStandard\Rules;
+
+use Nette\Utils\Strings;
+use PhpParser\Node;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\Stmt\ClassMethod;
+use PhpParser\NodeFinder;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\Rule;
+use PHPStan\Type\ThisType;
+
+/**
+ * @see \Symplify\CodingStandard\Tests\Rules\PreferredRawDataInTestDataProviderRule\PreferredRawDataInTestDataProviderRuleTest
+ */
+final class PreferredRawDataInTestDataProviderRule implements Rule
+{
+    /**
+     * @var string
+     */
+    public const ERROR_MESSAGE = 'Code configured at setUp() cannot be used in data provider. Move it to test() method';
+
+    /**
+     * @var string
+     * @see https://regex101.com/r/WaNbZ1/2
+     */
+    private const DATAPROVIDER_REGEX = '#\*\s+@dataProvider\s+(?<dataProviderMethod>.*)\n?#';
+
+    /**
+     * @var NodeFinder
+     */
+    private $nodeFinder;
+
+    public function __construct(NodeFinder $nodeFinder)
+    {
+        $this->nodeFinder = $nodeFinder;
+    }
+
+    public function getNodeType(): string
+    {
+        return ClassMethod::class;
+    }
+
+    /**
+     * @param ClassMethod $node
+     * @return string[]
+     */
+    public function processNode(Node $node, Scope $scope): array
+    {
+        $dataProviderMethodName = $this->matchDataProviderMethodName($node);
+        if ($dataProviderMethodName === null) {
+            return [];
+        }
+
+        $classMethod = $this->findDataProviderClassMethod($node, $dataProviderMethodName);
+        if ($classMethod === null) {
+            return [];
+        }
+
+        if ($this->isSkipped($classMethod, $scope)) {
+            return [];
+        }
+
+        return [self::ERROR_MESSAGE];
+    }
+
+    private function findDataProviderClassMethod(ClassMethod $classMethod, string $dataProviderMethod): ?ClassMethod
+    {
+        $class = $classMethod->getAttribute('parent');
+
+        /** @var ClassMethod[] $classMethods */
+        $classMethods = $this->nodeFinder->findInstanceOf($class, ClassMethod::class);
+        foreach ($classMethods as $classMethod) {
+            if ((string) $classMethod->name !== $dataProviderMethod) {
+                continue;
+            }
+
+            return $classMethod;
+        }
+
+        return null;
```

Comment:
I just realized this.... There is this method in `ClassLike` you can use.

 `$class->getMethod($methodName)` instead of all this

https://github.com/nikic/PHP-Parser/blob/d520bc9e1d6203c35a1ba20675b79a051c821a9e/lib/PhpParser/Node/Stmt/ClassLike.php#L100

Reply:
I'll process it along with other changes I do, don't worry